autoextract/aio/client.py

```diff
@@ -34,11 +37,103 @@ class Result(List[Dict]):
     response_stats: Optional[List[ResponseStats]] = None
 
 
+class RequestProcessor:
+    """Help keeping track of query results and errors between retries.
+
+    After initializing your Request Processor,
+    you may use it for just a single or for multiple requests.
+
+    This class is especially useful because it stores
+    successful queries to avoid repeating them when retrying requests.
+    """
+
+    def __init__(self, query: Query, max_retries: int = 0):
+        """Reset temporary data structures and initialize them"""
+        self._reset()
+        self.pending_queries = query_as_dict_list(query)
+        self._max_retries = max_retries
+        self._complete_queries: List[Dict] = list()
+
+    def _reset(self):
+        """Clear temporary variables between retries"""
+        self.pending_queries: List[Dict] = list()
+        self._retriable_queries: List[Dict] = list()
+        self._retriable_query_exceptions: List[Dict] = list()
+
+    def _enqueue_error(self, query_result, query_exception):
+        """Enqueue Query-level error.
+
+        Enqueued errors could be:
+            - used in combination with successes with `get_latest_results`
+            - retried using `pending_requests`
+        """
+        self._retriable_queries.append(query_result)
+        self._retriable_query_exceptions.append(query_exception)
+
+        user_query = query_result["query"]["userQuery"]
+        self.pending_queries.append(user_query)
+
+    def get_latest_results(self):
+        """Get latest results (errors + successes).
+
+        This method could be used to retrieve results
+        when an exception is raised while processing results.
+        """
+        return self._complete_queries + self._retriable_queries
+
+    def process_results(self, query_results):
+        """Process query results.
+
+        Return successful queries and also failed ones.
+
+        If `self._handle_exceptions` is True,
```

Comment:
Is this comment up to date?

Reply:
It's supposed to be `_max_retries`. Updated.